azext_iot/iothub/providers/device_identity.py

```diff
@@ -0,0 +1,391 @@
+            parent_id = device_to_parent_dict[device_id]
+            parent_scope = assembled_device_dict[parent_id].device_scope
+            device.parent_scopes = [parent_scope]
+            self.service_sdk.devices.create_or_update_identity(
+                id=device_id, device=device, if_match="*"
+            )
+
+        # update edge config / set-modules
+        devices_config_iterator = (
+            tqdm(config.devices, desc="Setting edge module content")
+            if visualize
+            else config.devices
+        )
+        for device_config in devices_config_iterator:
+            device_id = device_config.device_id
+            deployment_content = device_config.deployment
+            if deployment_content:
+                self.service_sdk.configuration.apply_on_edge_device(
+                    id=device_id, content=deployment_content
+                )
+
+    def _process_nested_edge_config_file_content(
+        self, content: dict
+    ) -> NestedEdgeConfig:
+        """
+        Process edge config file schema dictionary
+        """
+        # TODO version / schema validation
+        version = content["config_version"]
+        hub_config = content["iothub"]
+        devices_config = content["edgedevices"]
+        auth_value = hub_config["authentication_method"]
+        if auth_value not in ["symmetric_key", "x509_certificate"]:
+            raise InvalidArgumentValueError(
+                "Invalid authentication_method in edge config file, must be either symmetric_key or x509_certificate"
+            )
+        device_authentication_method = (
+            DeviceAuthType.shared_private_key.value
+            if auth_value == "symmetric_key"
+            else DeviceAuthType.x509_thumbprint.value
+        )
+        all_devices = []
+
+        def _process_edge_config_device(device: dict, parent_id=None):
+            device_id = device.get("device_id", None)
+            if not device_id:
+                raise InvalidArgumentValueError(
+                    "A device parameter is missing required attribute 'device_id'"
+                )
+            deployment = device.get("deployment", None)
+            if deployment:
+                deployment = self._try_parse_valid_deployment_config(deployment)
+            child_devices = device.get("child", [])
+            device_obj = NestedEdgeDeviceConfig(
+                device_id=device_id, deployment=deployment, parent=parent_id
+            )
+            all_devices.append(device_obj)
+            for child_device in child_devices:
+                _process_edge_config_device(child_device, parent_id=device_id)
+
+        for device in devices_config:
+            _process_edge_config_device(device)
+        return NestedEdgeConfig(
+            version=version,
+            auth_method=device_authentication_method,
+            devices=all_devices,
+        )
+
+    def assemble_nargs_to_dict(self, hash_list: List[str]) -> Dict[str, str]:
```

Comment:
theoretically we should be able to leverage this utility function across service areas (such as between here and adu), we can think about generic location to put it in

Reply:
Ok, I'm going to move this PR into a feature branch, but for now I've pulled in your changes from this recent PR: #586